### DISCUSSION TITLE
/login endpoint is throwing 403

### DISCUSSION THREAD

**Discussion by Trinankur** [Project-Books/book-project#843] [Q&A] [answered]
Hi,

I have followed all the mentioned steps in the README.md file and have started the database `docker-compose up -d mysql phpmyadmin` and server `java -jar target/book-project-0.2.0.jar` successfully. Anyhow, I am not able to log in using the test user credentials mentioned in the README.md file. I have tried from postman also (thought might be frontend issue) but still no luck, it's throwing 403 forbidden. Not sure if I have missed something. Could you please help me with that?

**Comment by knjk04** [comment, marked as answer]
Hi, could you please show a screenshot of your request in postman?

**Reply by Trinankur** [reply]
Below are the screenshots of the request:

**Response Body:**
![image](https://user-images.githubusercontent.com/43852092/137634988-d950e644-9bd6-44d5-a4a7-162316e6cb1f.png)

**Response Headers**
![image](https://user-images.githubusercontent.com/43852092/137635065-dc64186d-ab78-4e7f-a6a9-e7b9af06115b.png)
![image](https://user-images.githubusercontent.com/43852092/137635086-60e0b97a-7f54-42d7-a22b-1342426b2a93.png)

**Reply by bayang** [reply]
I had the same issue, the cause is e20b14e447bb7fd16d6cdfa50b1df0d03a7a7c30
Re adding csrf().disable() fixes the problem

**Reply by knjk04** [reply]
@Trinankur Does the above solution work for you?

**Reply by Trinankur** [reply]
Thanks @bayang and @knjk04, it worked!

**Reply by gweesin** [reply]
I also encountered this problem(main branch), and I also set the code of `csrf().disable()`.
I run the frontend and backend program alone.

```
// in SecurityConfiguration.java
@Override
protected void configure(HttpSecurity http) throws Exception {
    http.cors()
        .and()
        .csrf().disable()
        .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
        .and()
        .addFilter(new JwtUsernamePasswordAuthFilter(authenticationManager(), jwtConfig,
                secretKey))
        .addFilterAfter(new JwtTokenVerifier(secretKey, jwtConfig), JwtUsernamePasswordAuthFilter.class)
        .authorizeRequests()
        .antMatchers(HttpMethod.POST, Mappings.USER).permitAll()
        .anyRequest()
        .authenticated();
}
```

I just modify the config file(application.properties) like this:

```
server.port=8082

spring.jpa.hibernate.ddl-auto=validate
spring.jpa.properties.hibernate.dialect = org.hibernate.dialect.PostgreSQLDialect
spring.datasource.url = jdbc:postgresql://localhost:5432/book_project_db
spring.datasource.username = postgres
spring.datasource.password = 123456

spring.data.rest.base-path=/api

application.jwt.secretKey=4&lTw7H@VWQtYC$3gh9DD9cReS7ehZ&tL9^wq@ODH@#8TCIZuRkcYE2%Qg$3X$!2
application.jwt.tokenPrefix=Bearer 
application.jwt.tokenExpirationAfterDays=7
```

See the postman:

![image](https://user-images.githubusercontent.com/42909374/158059724-6b0b1053-e969-4050-99fb-a31fe9f951c4.png)

See the frontend:

![image](https://user-images.githubusercontent.com/42909374/158059766-a66a7587-b652-47a4-9bc5-c922d11d4864.png)

And very confused that in addition to the `/login` interface, all other requests are for port 3000.😭 like this: 

![image](https://user-images.githubusercontent.com/42909374/158059793-23f7f224-c436-4f90-9ffb-87d80eb3e2c2.png)

Finally, I'm sorry for so many questions. I hope I can learn and become a part of you!

**Reply by knjk04** [reply]
@c-g-x We're happy to help with this. Could you please create a new GitHub discussion for this, as the fix will be different and this one has already been resolved? Thanks!

**Reply by gweesin** [reply]
@knjk04 alright, thank you for your help!
